models/tiegcm/shell_scripts_gadi/assimilate.csh

```diff
@@ -0,0 +1,211 @@
+# STEP 4: [OPTIONAL] prepare for DART INFLATION 
+# This stages the files that contain the inflation values.
+# The inflation values change through time and should be archived.
+#
+# This file is only relevant if 'inflation' is turned on -
+# i.e. if inf_flavor(:) /= 0 AND inf_initial_from_restart = .TRUE.
+#
+# The strategy is to use the LATEST (last) inflation file, which is
+# always available in the contents of the pointer file.
+# ------------------------------------------------------------------------------
+#
+# Only supporting prior state-space inflation for now.
+
+# If the inflation pointer file does not exist, we need to make one.
+# This should only happen on the first assimilation cycle.
+
+if (! -e priorinf_pointer_file.txt ) then
+   if ( -e prior_inflate_ics ) then # run fill_inflation_restart.f90 
+      # ./fill_inflation_restart
+      $MOVE prior_inflate_ics prior_inflate_ics.$TIMESTAMP
+      echo "prior_inflate_ics.$TIMESTAMP" >! priorinf_pointer_file.txt
+   else
+      echo "ERROR: inflation staging failed."
+      exit 4
+   endif
+endif
+
+# Link the inflation file (output_priorinf_[mean,sd].DATE.nc) from
+# the previous assimilation cycle to be used as input for the current
+# assimilation. These are simply read, so linking is sufficient.
+
+set inflation_file = `head -n 1 priorinf_pointer_file.txt`
+
+${LINK} -f $inflation_file prior_inflate_ics
+
+# ------------------------------------------------------------------------------
+# STEP 5: assimilate the ensemble of new states.
+#
+# The stage_experiment script linked all the TIE-GCM files in the instance
+# directories to the expected names required by filter. Should be good to go.
+# ------------------------------------------------------------------------------
+
+${MPI_RUN_CMD} ./filter || exit 3
+
+if ( ! -e obs_seq.final ) then
+   echo "ERROR: filter did not finish successfully - no obs_seq.final"
+   exit 5
+endif
+
+# ------------------------------------------------------------------------------
+# STEP 6: Tag the output with the valid time of the model state.
+# Update the inflation pointer file with the most recent file.
+# ------------------------------------------------------------------------------
+
+${MOVE} -v obs_seq.final     obs_seq.final.$TIMESTAMP
+${MOVE} -v namelist_update namelist_update.$TIMESTAMP
+
+if ( -e prior_inflate_restart ) then
+   ${MOVE} -v prior_inflate_restart prior_inflate_restart.$TIMESTAMP
+   echo "prior_inflate_restart.$TIMESTAMP" >! priorinf_pointer_file.txt
+endif
+
+# Must copy each filter_restart.nnnn to the right place, update link
+# so that 'dart_to_model' gets what it needs.
+foreach RESTART ( filter_restart.???? )
+   set instance = $RESTART:e
+   set INSTANCE_DIRECTORY = `printf "instance_%04d" $instance`
+   cd $INSTANCE_DIRECTORY
+   ${MOVE} -v ../$RESTART  $RESTART.$TIMESTAMP
+   \unlink dart_restart
+   ${LINK} $RESTART.$TIMESTAMP dart_restart
+   cd ..
+end
```

Comment:
After this loop finishes, there should be a `dart_restart` in each of the instance directories.  Is this not happenning?